osu.Game.Rulesets.Taiko.Tests/Skinning/TestSceneDrawableTaikoMascot.cs

```diff
@@ -0,0 +1,155 @@
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Testing;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Judgements;
+using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Rulesets.Taiko.Objects.Drawables;
+using osu.Game.Rulesets.Taiko.UI;
+using osu.Game.Rulesets.UI.Scrolling;
+using osu.Game.Skinning;
+using osu.Game.Tests.Visual;
+
+namespace osu.Game.Rulesets.Taiko.Tests.Skinning
+{
+    [TestFixture]
+    public class TestSceneDrawableTaikoMascot : TaikoSkinnableTestScene
+    {
+        public override IReadOnlyList<Type> RequiredTypes => base.RequiredTypes.Concat(new[]
+        {
+            typeof(DrawableTaikoMascot),
+        }).ToList();
+
+        [Cached(typeof(IScrollingInfo))]
+        private ScrollingTestContainer.TestScrollingInfo info = new ScrollingTestContainer.TestScrollingInfo
+        {
+            Direction = { Value = ScrollingDirection.Left },
+            TimeRange = { Value = 5000 },
+        };
+
+        private readonly List<DrawableTaikoMascot> mascots = new List<DrawableTaikoMascot>();
+        private readonly List<SkinnableDrawable> skinnables = new List<SkinnableDrawable>();
+        private readonly List<TaikoPlayfield> playfields = new List<TaikoPlayfield>();
+
+        [Test]
+        public void TestStateTextures()
+        {
+            AddStep("Create mascot (idle)", () =>
+            {
+                skinnables.Clear();
+                SetContents(() =>
+                {
+                    var skinnable = getMascot();
+                    skinnables.Add(skinnable);
+                    return skinnable;
+                });
+            });
+
+            AddUntilStep("Wait for SkinnableDrawable", () => skinnables.Any(d => d.Drawable is DrawableTaikoMascot));
+
+            AddStep("Collect mascots", () =>
+            {
+                mascots.Clear();
+
+                foreach (var skinnable in skinnables)
+                {
+                    if (skinnable.Drawable is DrawableTaikoMascot mascot)
+                        mascots.Add(mascot);
+                }
+            });
+
+            AddStep("Clear state", () => setState(TaikoMascotAnimationState.Clear));
+
+            AddStep("Kiai state", () => setState(TaikoMascotAnimationState.Kiai));
+
+            AddStep("Fail state", () => setState(TaikoMascotAnimationState.Fail));
+        }
+
+        private void setState(TaikoMascotAnimationState state)
+        {
+            foreach (var mascot in mascots)
+            {
+                if (mascot == null)
+                    continue;
```

Comment:
Is this even possible? I'm pretty sure `null` should fail the `is DrawableTaikoMascot mascot` check. ([see docs](https://docs.microsoft.com/pl-pl/dotnet/csharp/language-reference/keywords/is#pattern-matching-with-is))

Reply:
To be honest, I'm surprised by this myself. I have no idea how Drawables can turn themselves to null.